src/sas/qtgui/Calculators/GenericScatteringCalculator.py

```diff
@@ -262,12 +267,21 @@ def check_value(self):
         if text_edit.text():
             value = float(str(text_edit.text()))
             if text_edit == self.txtQxMax:
-                if value <= 0 or value > 1000:
+                try:
+                    max_q = numpy.pi / (max(self.sld_data.xstepsize, self.sld_data.ystepsize, self.sld_data.zstepsize) )                   
+                except:
+                    max_q = 1000
```

Comment:
You don't want to use bare except statements (here and elsewhere): they block program shutdown signals such as Ctrl-C.

At minimum use "except Exception". Better still, only trap exceptions that you anticipate and can deal with. In this case the expression should only fail is if sld_data is not available or step sizes are not defined, both of which will raise `AttributeError`.

Wouldn't it be better to trap missing sld_data, etc., earlier? Conversely, if max_q/max_step are failing because the information isn't available yet, do they then get updated when the information becomes available? Maybe the solution is to avoid assigning max_q/max_step until they are needed in the calculation?

Reply:
The reader now catches missing values in the data, such that the input should be reasonable. All other (unavoidable) instances of except have been changes to "except Exception".

Reply:
This `except:` is still unchanged? Yes the chances of an interrupt within this try block are vanishingly small, but still non-zero.

Also, unnecessary parens around max().

Reply:
This is resolved with the last commit c949983 "catch except cases already at the inputs".